docs/website/docs/dlt-ecosystem/verified-sources/slack.md

```diff
@@ -0,0 +1,264 @@
+   [duckdb](../destinations/duckdb.md) as the [destination](../destinations).
+
+1. If you'd like to use a different destination, simply replace `duckdb` with the name of your
+   preferred [destination](../destinations).
+
+1. After running this command, a new directory will be created with the necessary files and
+   configuration settings to get started.
+
+For more information, read the
+[Walkthrough: Add a verified source.](../../walkthroughs/add-a-verified-source)
+
+### Add credentials
+
+1. In the `.dlt` folder, there's a file called `secrets.toml`. It's where you store sensitive
+   information securely, like access tokens. Keep this file safe. Here's its format for service
+   account authentication:
+
+   ```toml
+   [sources.slack]
+   access_token = "Please set me up!" # please set me up!
+   ```
+
+1. Copy the user Oauth token you [copied above](#grab-user-oauth-token).
+
+1. Finally, enter credentials for your chosen destination as per the [docs](../destinations/).
+
+## Run the pipeline
+
+1. Before running the pipeline, ensure that you have installed all the necessary dependencies by
+   running the command:
+
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+1. You're now ready to run the pipeline! To get started, run the following command:
+
+   ```bash
+   python3 slack_pipeline.py
+   ```
+
+1. Once the pipeline has finished running, you can verify that everything loaded correctly by using
+   the following command:
+
+   ```bash
+   dlt pipeline <pipeline_name> show
+   ```
+
+   For example, the `pipeline_name` for the above pipeline example is `slack`, you
+   may also use any custom name instead.
+
+   For more information, read the [Walkthrough: Run a pipeline](../../walkthroughs/run-a-pipeline).
+
+## Sources and resources
+
+`dlt` works on the principle of [sources](../../general-usage/source) and
+[resources](../../general-usage/resource).
+
+### Source `slack_source`
+
+It retrieves data from Slack's API and returns resources conversations, conversations_history, and access_logs.
+
+```python
+@dlt.source(name="slack", max_table_nesting=2)
+def slack_source(
+    page_size: int = MAX_PAGE_SIZE,
+    access_token: str = dlt.secrets.value,
+    start_date: Optional[TAnyDateTime] = DEFAULT_START_DATE,
+    end_date: Optional[TAnyDateTime] = None,
+    selected_channels: Optional[List[str]] = dlt.config.value,
+) -> Iterable[DltResource]:
+```
+
+`page_size`: Maximum items per page (default: 1000).
+
+`access_token`: OAuth token for authentication.
+
+`start_date`: Range start. (default: January 1, 2000).
+
+`end_date`: Range end.
+
+`selected_channels`: Channels to load; defaults to all if unspecified.
+
+### Resource `channels_resource`
+
+This function yields all the channels as `dlt` resource.
+
+
+```python
+@dlt.resource(name="channels", primary_key="id", write_disposition="replace")
+def channels_resource() -> Iterable[TDataItem]:
+   
+    yield from channel
```

Comment:
I think that if you are adding this you should add the lines above it, as the `channel` variable is defined above, but I am not sure about it.